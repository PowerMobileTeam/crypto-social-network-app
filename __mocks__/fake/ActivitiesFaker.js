
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

