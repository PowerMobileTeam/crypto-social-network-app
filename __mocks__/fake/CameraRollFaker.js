
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

