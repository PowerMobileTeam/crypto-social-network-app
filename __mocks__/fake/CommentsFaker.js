
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

