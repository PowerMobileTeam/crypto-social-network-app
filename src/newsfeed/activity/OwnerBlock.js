import React, {
  PureComponent
} from 'react';

import {
  NavigationActions
} from 'react-navigation';

import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  View
} from 'react-native';

import {
  MINDS_CDN_URI
} from '../../config/Config';

/**
 * Owner Block Component
 */
export default class OwnerBlock extends PureComponent {

  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    // only active if receive the navigation property
    if (this.props.navigation) {
      this.props.navigation.navigate('Channel', { guid:this.props.entity.ownerObj.guid, entity: this.props.entity.ownerObj });
    }
  }

  /**
   * Navigate To group
   */
 _navToGroup = () => {
    if (this.props.navigation) {
      this.props.navigation.navigate('GroupView', { group: this.props.entity.containerObj });
    }
  };

  get group() {
    if(!this.props.entity.containerObj)
      return null;

    return (
      <TouchableOpacity onPress={this._navToGroup} style={styles.groupContainer}>
        <Text style={styles.groupName} lineBreakMode='tail' numberOfLines={1}>
          > { this.props.entity.containerObj.name }
        </Text>
      </TouchableOpacity>
    );
  }

  /**
   * Render
   */
  render() {
    const channel = this.props.entity.ownerObj;
    const rightToolbar = this.props.rightToolbar||null;

    const avatarSrc = channel.getAvatarSource();

    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={this._navToChannel}>
          <Image source={avatarSrc} style={styles.avatar}/>
        </TouchableOpacity>
        <View style={styles.body}>
          <View style={styles.nameContainer}>
            <TouchableOpacity onPress={this._navToChannel}>
              <Text style={styles.username}>
                { channel.username }
              </Text>
            </TouchableOpacity>
            { this.group }
          </View>
          {this.props.children}
        </View>
        {rightToolbar}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    padding: 8,
    alignItems: 'center',
    flexDirection: 'row',
  },
  avatar: {
    height: 46,
    width: 46,
    borderRadius: 23,
    borderWidth: 0.5,
    borderColor: '#EEE',
  },
  body: {
    marginLeft: 8,
    paddingRight: 36,
    flexWrap: 'wrap',
    flex: 1,
  },
  nameContainer: {
    flexDirection: 'row',
  },
  username: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#444',
  },
  groupContainer: {
    marginLeft: 4,
    flex: 1,
  },
  groupName: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#888',
  }
});
