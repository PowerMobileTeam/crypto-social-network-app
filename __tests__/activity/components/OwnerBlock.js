import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import OwnerBlock from '../../../src/newsfeed/activity/OwnerBlock';

describe('Owner component', () => {

  let screen;
  beforeEach(() => {

    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <OwnerBlock entity={activityResponse.activities[0]} navigation={navigation} rightToolbar={null}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have Touchableopacity', async () => {
    screen.update();
    expect(screen.find(TouchableOpacity)).toHaveLength(2);
  });

  it('should _navToChannel on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = { 
      navigate: jest.fn() 
    };
    let entity = activityResponse.activities[0];
    entity.containerObj = { guid: 'guidguid' };
    screen = shallow(
      <OwnerBlock entity={entity} navigation={navigation} rightToolbar={null}/>
    );
    screen.update()
    let render = screen.dive();
    let touchables = render.find('TouchableOpacity');
    touchables.at(0).props().onPress();
    jest.runAllTimers();

    expect(navigation.navigate).toHaveBeenCalledWith('Channel', {'entity': entity.ownerObj, 'guid': entity.ownerObj.guid});


    expect(screen.find(TouchableOpacity)).toHaveLength(3);

  });


  it('should nav to groups on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = { 
      navigate: jest.fn() 
    };
    let entity = activityResponse.activities[0];
    entity.containerObj = { guid: 'guidguid' };
    screen = shallow(
      <OwnerBlock entity={entity} navigation={navigation} rightToolbar={null}/>
    );
    screen.update()
    let render = screen.dive();
    let touchables = render.find('TouchableOpacity');


    expect(screen.find(TouchableOpacity)).toHaveLength(3);
    //group touchable
    touchables.at(2).props().onPress();
    jest.runAllTimers();

    expect(navigation.navigate).toHaveBeenCalled();

  });


});
