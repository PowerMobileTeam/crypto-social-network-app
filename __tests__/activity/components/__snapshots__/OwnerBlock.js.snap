// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Owner component renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Image
      source={
        Object {
          "source": "http://thisisaurl",
        }
      }
      style={
        Object {
          "borderColor": "#EEE",
          "borderRadius": 23,
          "borderWidth": 0.5,
          "height": 46,
          "width": 46,
        }
      }
    />
  </TouchableOpacity>
  <Component
    style={
      Object {
        "flex": 1,
        "flexWrap": "wrap",
        "marginLeft": 8,
        "paddingRight": 36,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#444",
              "fontFamily": "Roboto",
              "fontWeight": "bold",
            }
          }
        />
      </TouchableOpacity>
    </Component>
  </Component>
</Component>
`;
